app/utils/results.py

```diff
@@ -223,3 +223,101 @@ def add_ip_card(html_soup: BeautifulSoup, ip: str) -> BeautifulSoup:
         # Inserting the element
         ref_element.insert_before(ip_tag)
     return html_soup
+
+
+def check_currency(response: str) -> dict:
+    """Check whether the results have currency conversion
+
+    Args:
+        response: Search query Result
+
+    Returns:
+        dict: Consists of currency names and values
+
+    """
+    soup = BeautifulSoup(response, 'html.parser')
+    currency_link = soup.find('a', {'href': 'https://g.co/gfd'})
+    if currency_link and currency_link.text.lower() == 'disclaimer':
+        for i in range(4):
+            currency_link = currency_link.parent
+        currency1 = currency_link
+        for i in range(5):
+            currency1 = currency1.parent
+        for i in range(2):
+            currency1 = currency1.previous_sibling
```

Comment:
I think looking for a specific class name would actually be a better approach -- if we switch to traversing up N number of parents, and the layout for the conversion card returned by Google gets changed at some point, it'll be more effort to determine how deep the child element is nested. As long as we keep it to one class name, it should be easy enough to maintain in the future. 

Something like (pseudocode, not tested):

```py
while currency_link.parent and 'ZINbbc' not in currency_link.attrs['class']:
    currency_link = currency_link.parent
```